sunpy/image/slit.py

```diff
@@ -0,0 +1,129 @@
+import sunpy
+import sunpy.map
+import sunpy.cm
+import sunpy.wcs
+
+import numpy as np
+import astropy.wcs
+import astropy.io.fits as fits
+import astropy.units as u
+
+import scipy.ndimage as snd
+import matplotlib.pyplot as plt
+from glob import glob
+
+
+
+def slit(in_files, xy1, xy2):
+    """
+    Returns an array with intensity along the slit on the y axis and time 
+    along x.
+    
+    Parameters    
+    ----------
+    in_files : `sunpy.map.MapCube`
+        A mapcube of the images you want to perform the slit analysis on. 
+        Usually with x and y as space, and z as time.
+    xy1 : List of x and y coordinates in `[x1, y1]`
+        The x and y coordinates of the beginning of the slit. May be either 
+        pixel coordinates or astropy units.
+    xy2 : List of x and y coordinates
+        The x and y coordinates of the end of the slit. May be either 
+        pixel coordinates or astropy units.
+
+        
+    Returns
+    -------
+    out : numpy array
+        A numpy array with intenisty in y and time in x.
+
+    """
+    
+    # check the attributes of the coordinates
+    if ((isinstance(xy1[0] and xy1[1], u.Quantity) and isinstance(xy2[0] and xy2[1], u.Quantity)) or
```

Comment:
related to above. You have this setup at the moment to take in two lists of Quantities, which while ok, is not the same as similar functions. Therefore I would suggest you take in two Quantity objects for the two coordinates, and then when you check type here you can just do:

```python
if (isinstance(xy1 and xy2, u.Quantity) or hasattr(xy1 and xy2, 'unit')):
```